addons/note/models/res_users.py

```diff
@@ -0,0 +1,17 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, models
+
+
+class ResUsers(models.Model):
+    _inherit = 'res.users'
+
+    @api.model
+    def create(self, data):
+        user = super(ResUsers, self).create(data)
+        if user.has_group('base.group_user'):
+            for n in range(5):
+                stage_id = self.env.ref('note.note_stage_%02d' % (n,), raise_if_not_found=False)
+                stage_id.sudo().copy(default={'user_id': user.id})
```

Comment:
stage_id may contain 'False', hence, you need to check for it's value before copying it.